cmd/kubeadm/app/util/pubkeypin/pubkeypin.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pubkeypin provides primitives for x509 public key pinning in the
+// style of RFC7469.
+package pubkeypin
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+)
+
+// Set is a set of pinned x509 public keys.
+type Set struct {
+	sha256Hashes [][]byte
```

Comment:
+1, will fix. I originally had this code supporting other hash types/lengths (truncated SHA-256/128 and SHA-256 encoded as base58). In the end I just left plain hex-encoded SHA-256 so this doesn't make a lot of sense.